orangecontrib/text/widgets/owtwitter.py

```diff
@@ -0,0 +1,275 @@
+import functools
+
+from PyQt4 import QtGui, QtCore
+from PyQt4.QtGui import QComboBox, QDialog
+
+from Orange.widgets import gui
+from Orange.widgets.settings import Setting
+from Orange.widgets.widget import OWWidget
+
+from orangecontrib.text import twitter
+from orangecontrib.text.corpus import Corpus
+from orangecontrib.text.widgets.ownyt import _i
+from orangecontrib.text.languge_codes import lang2code
+
+
+class APICredentialsDialog(QDialog):
+    size_hint = (470, 100)
+    widgets_min_width = 380
+
+    def __init__(self, parent, windowTitle="Twitter API credentials"):
+        super().__init__(parent, windowTitle=windowTitle)
+
+        self.parent = parent
+
+        self.setLayout(QtGui.QVBoxLayout())
+        self.layout().setMargin(10)
+        self.mainArea = gui.widgetBox(self)
+
+        self.layout().addLayout(self.setup_main_layout())
+
+        # Buttons
+        self.button_box = gui.widgetBox(self, orientation=QtCore.Qt.Horizontal)
+        gui.button(self.button_box, self, "Check", self.check_api_key)
+        gui.button(self.button_box, self, "Cancel", self.reject)
+        gui.button(self.button_box, self, "OK", self.accept)
+
+        # Label
+        # self.label_box = gui.widgetBox(self, orientation=QtCore.Qt.Horizontal)
+        self.api_key_check_label = QtGui.QLabel(text='')
+        self.api_key_check_label.setAlignment(QtCore.Qt.AlignCenter)
+        self.layout().addWidget(self.api_key_check_label)
+
+        self.check_api_key()
+
+    def setup_main_layout(self):
+        layout = QtGui.QFormLayout()
+
+        # Key combo box.
+        self.key_combo = QComboBox(self)
+        self.key_combo.setEditable(True)
+        self.key_combo.activated[int].connect(self.key_selected)
+        self.key_combo.setMinimumWidth(self.widgets_min_width)
+        layout.addRow('Key', self.key_combo)
+
+        self.secret_line_edit = QtGui.QLineEdit(self)
+        self.secret_line_edit.editingFinished.connect(self.secret_changed)
+        self.secret_line_edit.setMinimumWidth(self.widgets_min_width)
+        layout.addRow('Secret', self.secret_line_edit)
+        self.set_key_list()
+        return layout
+
+    def sizeHint(self):
+        return QtCore.QSize(*self.size_hint)
+
+    @property
+    def key(self):
+        return self.recent_keys[0] if self.recent_keys else None
+
+    @property
+    def recent_keys(self):
+        return self.parent.recent_api_keys
+
+    def set_key_list(self):
+        self.key_combo.clear()
+        if self.recent_keys:
+            for key in self.recent_keys:
+                self.key_combo.addItem(key.consumer_key)
+            self.secret_line_edit.setText(self.recent_keys[0].consumer_secret)
+
+    def check_api_key(self):
+        key = twitter.Credentials(self.key_combo.currentText(),
+                                  self.secret_line_edit.text())
```

Comment:
Do you think it would be OK to add `.strip()` here? Sometimes I copy an extra space at the beginning or end.